lib/remote/apilistener.cpp

```diff
@@ -1526,8 +1526,12 @@ double ApiListener::CalculateZoneLag(const Endpoint::Ptr& endpoint)
 bool ApiListener::AddAnonymousClient(const JsonRpcConnection::Ptr& aclient)
 {
 	boost::mutex::scoped_lock lock(m_AnonymousClientsLock);
+	auto mac (GetMaxAnonymousClients());
 
-	if (GetMaxAnonymousClients() >= 0 && (long)m_AnonymousClients.size() + 1 > (long)GetMaxAnonymousClients())
+	if (mac < 0)
+		mac = ConfigType::GetObjectsByType<Endpoint>().size() * 2u;
```

Comment:
Why do we do this at runtime? Can we do this only once after compiling the configuration so that `GetMaxAnonymousClients()` always returns a sane value?